### PR TITLE
Ajax attachment card

### PR DESCRIPTION
**Pull request by nickbristow**

### CODE REVIEW

File: app/controllers/attachments_controller.rb [hunk 1]
```diff
@@ -2,24 +2,32 @@ class AttachmentsController < ApplicationController
   before_action ->{authorize proposal, :can_show!}, only: [:create, :show]
   before_action ->{authorize attachment}, only: [:destroy]
   rescue_from Pundit::NotAuthorizedError, with: :auth_errors
+  respond_to :js, :only => [:create, :destroy]
```
Comment (jessieay) [inline app/controllers/attachments_controller.rb]: new hash syntax please!

File: app/controllers/attachments_controller.rb [hunk 2]
```diff
@@ -2,24 +2,32 @@ class AttachmentsController < ApplicationController
...
-    if attachment.save
+    @attachment = proposal.attachments.build(attachments_params)
+    @attachment.user = current_user
+
+    if @attachment.save
       flash[:success] = "You successfully added a attachment"
-      DispatchFinder.run(proposal).deliver_attachment_emails(attachment)
+      respond_to do |format|
+        format.js
+        format.html {redirect_to proposal}
```
Comment (jessieay) [inline app/controllers/attachments_controller.rb]: spaces within curly braces por favor

File: app/controllers/attachments_controller.rb [hunk 3]
```diff
@@ -2,24 +2,32 @@ class AttachmentsController < ApplicationController
...
+      end
+      DispatchFinder.run(proposal).deliver_attachment_emails(@attachment)
     else
       flash[:error] = attachment.errors.full_messages
```
Comment (jessieay) [inline app/controllers/attachments_controller.rb]: need a redirect here since we removed it below

File: app/views/attachments/create.js.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+$("div.card-for-attachments div.attachments").html('<%=j render partial: 'proposals/details/attachment_list', locals: { proposal: @attachment.proposal } %>');
```
Comment (jessieay) [inline app/views/attachments/create.js.erb]: what if there is an error? do we show that someone via the js template?

**Comment by nickbristow** [comment]
Code climate issues are related to methods introduced before this PR. 

Feel free to ignore or give comments on how to make them smaller.

**Comment by nickbristow** [comment]
@pkarman @rememberlenny 

I want to split the dropzone, handling the JS error messages, and restyling the upload button out into a separate PR. 

This PR focuses on getting the functionality down. 

Ready for review. 

screenshot_4_25_16__7_09_pm

**Comment by rememberlenny** [comment]
@nickbristow Looks like merge conflict

**Comment by nickbristow** [comment]
There we go @rememberlenny

File: app/views/proposals/details/_attachment.html.haml [hunk 5]
```diff
@@ -5,7 +5,7 @@
         .card-head.column
           %h2
             Attachments
-            = modify_client_button(@proposal, "<span>Attach File</span>".html_safe, "button secondary attach-icon fr large")
+            = #modify_client_button(@proposal, "<span>Attach File</span>".html_safe, "button secondary attach-icon fr large")
```
Comment (rememberlenny) [inline app/views/proposals/details/_attachment.html.haml]: @nickbristow Should this be deleted?

**Pull request merged**